content/posts/tracing-from-0-to-100.md

```diff
@@ -0,0 +1,131 @@
+
+It seams we are stuck in an endless loop of trying to get everyone to agree on a standard, and then trying to get everyone to implement it correctly. This is where OpenTelemetry comes in. It provides a standard way to define telemetry data from your applications, and it provides libraries for most programming languages to make it easy to implement.
+
+## The first steps
+
+Even though OpenTelemetry does a lot of the heavy lifting for you, it is still a complex system with many moving parts. Did you know that OpenTelemetry is the fastest growing project in the Cloud Native Computing Foundation (CNCF)? It has an even steeper adoption curve than Kubernetes had in the early days!
+
+In order to get started with OpenTelemetry, you need two things:
+
+1. A place to store (and visualize) the telemetry data
+1. Convincing your developers that it is worth their time to instrument their applications
+
+Doesn't sound too hard, right? Let's start with the storage backend.
+
+OpenTelemetry is a vendor-neutral project, so you can choose any storage backend you like. The most popular choices are Jaeger, Zipkin, and Tempo. We chose [Grafana Tempo][grafana-tempo] because it is a scalable, cost-effective, and open source solution that integrates seamlessly with Grafana that we already use for metrics and dashboards.
+
+[grafana-tempo]: https://grafana.com/oss/tempo/
+![Grafana Tempo](../images/opentelemetry-tempo.png)
+
+We have written extensively about how to get started with Grafana Tempo in our [documentation][nav-tempo] as well as a reference guide for the query language used in Tempo called [TraceQL][nav-traceql].
+
+You can send OpenTelemetry data directly to Tempo, but the recommended way is to use an [OpenTelemetry Collector][otel-collector]. The Collector can receive data from multiple sources, process it, and send it to multiple destinations. This makes it easy to add new sources or destinations without changing your application configuration.
+
+Installing things like this in a Kubernetes cluster is something the NAIS team have done for the better part of ten years, so we had no problem setting up the Collector and connecting it to Tempo in our clusters, we run one Tempo instance for each environment (dev and prod) accessible from a global Grafana instance.
+
+The hard part was getting the developers to instrument their applications...
+
+[nav-traceql]: https://doc.nais.io/reference/observability/tracing/traceql/
+[nav-tempo]: https://docs.nais.io/how-to-guides/observability/tracing/tempo/
+[otel-collector]: https://opentelemetry.io/docs/collector/
+
+## Instrumenting the applications
+
+From the very beginning, we knew that the key to success was to make it as easy as possible for the developers to instrument their applications. With one 1.600 applications in production, we couldn't afford to spend weeks or months on each one. A solution that *required* manual configuration for each application was a non-starter.
```

Comment:
Et annet punkt for å få til adopsjon er i tillegg til å senke terskelen, at de er selvmotivert. Forstår og eller ser verdien av å gjøre dette. Kanskje også det med at du kan gå fra logg til trace til metrikk i ett og samme verktøy er 🙌